Backend/Repository/RoasterRepository.cs

```diff
@@ -0,0 +1,152 @@
+using Backend.Common;
+using Backend.Common.Params;
+using Backend.Data;
+using Backend.DTOs;
+using Backend.Entities;
+using Backend.Interfaces.Repository;
+using Microsoft.EntityFrameworkCore;
+
+namespace Backend.Repository;
+
+public class RoasterRepository(DataContext context) : BaseRepository<Roaster>(context), IRoasterRepository
+{
+  public async Task<RoasterDto?> GetRoasterByIdAsync(int id)
+  {
+    return await Context.Roasters
+      .Where(r => r.Id == id)
+      .Select(r => new RoasterDto
+      {
+        Id = r.Id,
+        Name = r.Name,
+        Alias = r.Alias,
+        Location = r.Location,
+        WebsiteUrl = r.WebsiteUrl,
+        Description = r.Description
+      })
+      .SingleOrDefaultAsync();
+  }
+
+  public async Task<PagedList<RoasterDto>> GetRoastersAsync(RoasterParams roasterParams)
+  {
+    var query = Context.Roasters
+      .Select(r => new RoasterDto
+      {
+        Id = r.Id,
+        Name = r.Name,
+        Alias = r.Alias,
+        Location = r.Location,
+        WebsiteUrl = r.WebsiteUrl,
+        Description = r.Description,
+      });
+
+    // TODO: For the text searches below, we should use tsvector
```

Comment:
remove this comment, decided to just not use tsvector for this, it also just wouldnt make sense since roasters are full of names that are likely not going to conform with the purposes of similar text search.